components/gantry/oneaxis/oneaxis.go

```diff
@@ -172,88 +172,54 @@ func (g *oneAxis) home(ctx context.Context, np int) error {
 	defer done()
 	// Mapping one limit switch motor0->limsw0, motor1 ->limsw1, motor 2 -> limsw2
 	// Mapping two limit switch motor0->limSw0,limSw1; motor1->limSw2,limSw3; motor2->limSw4,limSw5
+
+	var positionA, positionB float64
+	var err error
 	switch np {
+	// An axis with an encoder will encode the zero position, and add the second position limit
+	// based on the steps per length
+	case 0:
+		revPerLength := g.lengthMm / g.mmPerRevolution
+		positionA, err = g.motor.Position(ctx, nil)
+		if err != nil {
+			return err
+		}
+		positionB = positionA + revPerLength
 	// An axis with one limit switch will go till it hits the limit switch, encode that position as the
 	// zero position of the oneaxis, and adds a second position limit based on the steps per length.
 	case 1:
-		if err := g.homeOneLimSwitch(ctx); err != nil {
```

Comment:
question: Is there a reason for removing these function? 
Seems like home() is doing a lot. Having separate function calls made home() simpler than it was now.

Reply:
the original goal of the ticket was to combine all the homing logic into one function, but maybe we can go with the in-between solution? the first commit has the limit switch logic combined, and the encoder separate. thoughts on that @susmitaSanyal?

Reply:
We can actually simplify homeOneLimSwitch, homeTwoLimSwitch and homeEncoder functions by passing positionA/positionB as input. give the section below it's own function with 0.8 being a variable of this function.

```
	// Go backwards so limit stops are not hit.
	x := g.gantryToMotorPosition(0.8 * g.lengthMm)
	if err = g.motor.GoTo(ctx, g.rpm, x, nil); err != nil {
		return err
	}
```

and return better error messages in home() function. 
Let me know if this is makes sense.

Reply:
sure I can make this a helper function as well! would that mean I should leave it as is with this helper function or do you also think it would be clearer if we went back to having a function `homeEncoder` and a function `homeLimSwitch` like in the first commit on this PR